xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -751,6 +751,15 @@ void CActiveAESink::EnumerateOutputDevices(AEDeviceList &devices, bool passthrou
       if (passthrough && devInfo.m_deviceType == AE_DEVTYPE_PCM)
         continue;
 
+      // filters devices that should not be shown in the list
+      // of AUDIO DEVICES or AUDIO PASSTHROUGH DEVICES
+      // according to the capabilities informed by each sink
+      if (devInfo.m_onlyPassthrough && !passthrough)
```

Comment:
With MASK / ENUM this might be more extendable in the future. Though - yeah - currently we only have to distinguish between PCM and PT, as this is all the settings we show.

Reply:
See above: AE_DEVTYPE_PCM for PCM only (no PT) is already existing, therefore just the "onlyPassthrough" flag is needed. For PCM only devices, it's sufficient to set the device to type: AE_DEVTYPE_PCM

Reply:
> PCM = 0x1
> PT = 0x2
> 
> PCM | PT -> 0x3

This would surely require modifying all the sinks. I don't know the Linux situation at all. As it is now with the two bool flags the other untouched sinks will continue to work as they do now.

Reply:
See: https://github.com/xbmc/xbmc/pull/20856/files/5338f28ff37d7cbb157af4748d26bece31c648a3#diff-05e1b56df10155e0212d39b1d73fe66aa0712448de99811146e47688c9614b3bL751 <- if the type is PCM, which it can be if it has no PT capabilities, the m_pcmonly flag would not be needed.

Scratch the enum idea, it's not needed - as one boolean flag is enough.

Reply:
Line 751 does not filter DirectSound HDMI devices (Intel / Nvidia HDMI audio) then these DirectSound sinks gets listed on audio PT device list and can not be used for ALL passthrough formats.

> one boolean flag is enough.

not enough for commented reason.

Reply:
Obviously you have to set the DirectSound devices AE_DEVTYPE_PCM, see here: https://github.com/xbmc/xbmc/blob/5338f28ff37d7cbb157af4748d26bece31c648a3/xbmc/cores/AudioEngine/Sinks/AESinkDirectSound.cpp#L506

Reply:
But hey - long story short - if you want the pcm flag, it's okay ... might even be more explicit. Though keep in mind that kodi code already uses AE_DEVTYPE to distinguish PCM vs. PT - and the boolean flag would "double" that behaviour, cause actually AE_DEVTYPE_PCM would then be exactly the same as (AE_DEVTYPE_HDMI or AE_DEVTYPE_SPDIF) + m_pcmOnly set, therefore one could directly set the pcm only devices to AE_DEVTYPE_PCM.